src/hooks/useScreenWidth.ts

```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from "react"
+
+const useScreenWidth = () => {
+  const [width, setWidth] = useState<number>(window.innerWidth)
```

Comment:
Could this cause window is undefined issue ?

Reply:
Potentially if this is SSR, then - yes. And in this case we can use either if statement or "use client" flag.